packages/react/menu/src/Menu.tsx

```diff
@@ -739,14 +739,17 @@ const MenuSubTrigger = React.forwardRef((props, forwardedRef) => {
         data-state={getOpenState(context.open)}
         {...props}
         ref={composeRefs(forwardedRef, context.onTriggerChange)}
-        onPointerUp={composeEventHandlers(
-          props.onPointerUp,
-          whenTouchOrPen((event) => {
-            contentContext.onItemEnter(event);
```

Comment:
There are a few reasons why I removed this (please correct me if I'm wrong here 🙏):
- The `onItemEnter` is typed to accept `PointerEvent` only which I could have unioned but:
- The event seems to handle the grace logic which I don't think need here?
- A click event isn't an enter event

Reply:
> The event seems to handle the grace logic which I don't think was ever needed here because we don't handle hover/pointerdown -> move -> pointerup for touch/pen?

Yeah this is true in its current form. I think it was written this way to make an explicit distinction between the input methods. With this change it's less clear that the `onClick` is redundant on `mouse`, there's not much we can do about that though given the bug with `pointerUp` and the fact that we can't determine input type with `onClick`?

Perhaps raising a comment above the whole handler would help there?

Reply:
Comment added :+1: